lib/rules/avoid-leaking-state-in-ember-objects.js

```diff
@@ -52,6 +38,47 @@ module.exports = {
   DEFAULT_IGNORED_PROPERTIES,
 
   create(context) {
+    let emberImportAliasName;
+
+    const isEmberObjectCall = function (value) {
+      const callee = value.callee;
+
+      const isEmberObject = (utils.isMemberExpression(callee) &&
+        utils.isIdentifier(callee.object) && (callee.object.name === 'Ember' || callee.object.name === emberImportAliasName) &&
+        utils.isIdentifier(callee.property) && callee.property.name === 'Object');
+
+      const isEmberObjectCreate = (utils.isMemberExpression(callee) &&
+        utils.isIdentifier(callee.object) && callee.object.name === 'EmberObject' &&
+        utils.isIdentifier(callee.property) && callee.property.name === 'create');
+
+      return callee.name === 'Object' ||
+        callee.name === 'EmberObject' ||
+        isEmberObject ||
+        isEmberObjectCreate;
+    };
+
+    const isEmberArrayCall = function (value) {
+      const callee = value.callee;
+      return (callee.name === 'A') ||
+        (utils.isMemberExpression(callee) &&
+        utils.isIdentifier(callee.object) && (callee.object.name === 'Ember' || callee.object.name === emberImportAliasName) &&
+        utils.isIdentifier(callee.property) && callee.property.name === 'A');
+    };
+
+    const isAllowed = function (property) {
+      const value = property.value;
+
+      return ember.isFunctionExpression(value) ||
+        utils.isLiteral(value) ||
+        utils.isIdentifier(value) ||
+        (utils.isCallExpression(value) && !isEmberArrayCall(value) && !isEmberObjectCall(value)) ||
+        utils.isBinaryExpression(value) ||
+        utils.isTemplateLiteral(value) ||
+        utils.isTaggedTemplateExpression(value) ||
+        utils.isMemberExpression(value) ||
+        utils.isUnaryExpression(value);
+    };
```

Comment:
why did you move this into the function? this will redefine the function over and over for each call. since the function is not using any variables from the outer function I don't see a reason for moving it 🤔

Reply:
You are actually right! I only need emberImportAliasName which i can pass as an argument. 

For the rest of the code, i wonder if there's a better way of doing the checks. 

Thx for the feedback!